### PR TITLE
Switch to vitest

### PR DESCRIPTION
**Pull request by Shinigami92**
Related to https://github.com/dreamit-de/graphql-server-base/issues/73#issuecomment-1902044102

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,10 +39,10 @@
         "eslint-plugin-import": "2.29.1",
         "eslint-plugin-security": "2.1.0",
         "eslint-plugin-unicorn": "50.0.1",
-        "jest-html-reporters": "3.1.7",
         "prettier": "3.2.4",
-        "ts-jest": "29.1.1",
-        "typescript": "5.3.3"
+        "typescript": "5.3.3",
+        "vite-tsconfig-paths": "~4.3.1",
```
Comment (sgohlke) [inline package.json:44]: Is there a reason this dependency is not set to a fixed version? We use Dependabot to handle version updates, so it should be fine to set it to "4.3.1" and let Dependabot handle the updates.

**Review by sgohlke — COMMENTED** [review]
LGTM

File: package.json (hunk 1 above)
Comment (Shinigami92) [reply package.json:44]: I'm sorry, I have `save-prefix=~` in my `$HOME/.npmrc`
We should remove it, cause the repo prefers no lock files

Personally I work with `pnpm-lock.yaml`

Comment (Shinigami92) [inline package.json:44]:
```suggestion
        "vite-tsconfig-paths": "4.3.1",
```

**Review by sgohlke — APPROVED** [review]
lgtm

**Pull request merged**